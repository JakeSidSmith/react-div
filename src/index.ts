import React = require('react');

const originalCreateElement = React.createElement;

Object.defineProperty(React, 'createElement', {
  value: (...args: Parameters<typeof React.createElement>) => {
    const [type, ...rest] = args;
    const convertedType = typeof type === 'string' ? 'div' : type;

    return originalCreateElement.call(React, convertedType, ...rest);
  },
  configurable: true,
});
