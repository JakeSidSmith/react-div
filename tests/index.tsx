import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('react-div', () => {
  it("should override React's createElement function, and call the original with string elements swapped for divs", () => {
    const originalCreateElement = React.createElement;

    require('../src');

    expect(React.createElement).not.toBe(originalCreateElement);

    const TestComponent = () => null;

    expect(React.createElement('h1', {}, 'Hello, World!')).toEqual(
      originalCreateElement('div', {}, 'Hello, World!')
    );

    expect(React.createElement(TestComponent, {}, 'Hello, World!')).toEqual(
      originalCreateElement(TestComponent, {}, 'Hello, World!')
    );

    const App = () => (
      <main>
        <h1>Wow, look at this</h1>
        <p>
          None of these tags have <strong>any</strong> meaning any more!
        </p>
        <p>Open up the inspector and take a peek.</p>
      </main>
    );

    expect(renderer.create(<App />)).toMatchSnapshot();
  });
});
