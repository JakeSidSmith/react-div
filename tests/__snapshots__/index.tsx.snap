// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-div should override React's createElement function, and call the original with string elements swapped for divs 1`] = `
<div>
  <div>
    Wow, look at this
  </div>
  <div>
    None of these tags have 
    <div>
      any
    </div>
     meaning any more!
  </div>
  <div>
    Open up the inspector and take a peek.
  </div>
</div>
`;
